rectools/models/nn/transformer_base.py

```diff
@@ -839,19 +890,69 @@ def _recommend_i2i(
         )
 
     @property
-    def torch_model(self) -> TransformerBasedSessionEncoder:
+    def torch_model(self) -> TransformerTorchBackbone:
         """Pytorch model."""
         return self.lightning_model.torch_model
 
     @classmethod
     def _from_config(cls, config: TransformerModelConfig_T) -> tpe.Self:
         params = config.model_dump()
         params.pop("cls")
-        params["trainer"] = None
         return cls(**params)
 
     def _get_config(self) -> TransformerModelConfig_T:
         attrs = self.config_class.model_json_schema(mode="serialization")["properties"].keys()
         params = {attr: getattr(self, attr) for attr in attrs if attr != "cls"}
         params["cls"] = self.__class__
         return self.config_class(**params)
+
+    @classmethod
+    def _model_from_checkpoint(cls, checkpoint: tp.Dict[str, tp.Any]) -> tpe.Self:
+        """Create model from loaded Lightning checkpoint."""
+        model_config = checkpoint["hyper_parameters"]["model_config"]
+        loaded = cls.from_config(model_config)
+        loaded.is_fitted = True
+        dataset_schema = checkpoint["hyper_parameters"]["dataset_schema"]
+        dataset_schema = DatasetSchema.model_validate(dataset_schema)
+
+        # Update data preparator
+        id_map_schema = dataset_schema.items.id_map
+        item_external_ids = np.array(id_map_schema.external_ids, dtype=id_map_schema.dtype)
+        loaded.data_preparator.item_id_map = IdMap(item_external_ids)
+        loaded.data_preparator._init_extra_token_ids()  # pylint: disable=protected-access
+
+        # Init and update torch model and lightning model
+        torch_model = loaded._init_torch_model()
+        torch_model.construct_item_net_from_dataset_schema(dataset_schema)
+        loaded._init_lightning_model(torch_model, dataset_schema, model_config)
+        loaded.lightning_model.load_state_dict(checkpoint["state_dict"])
+
+        return loaded
+
+    def __getstate__(self) -> object:
+        if self.is_fitted:
+            with NamedTemporaryFile() as f:
+                if self.fit_trainer is None:
+                    raise TypeError("Model that was loaded from checkpoint cannot be saved without being fitted again")
+                self.fit_trainer.save_checkpoint(f.name)
+                checkpoint = Path(f.name).read_bytes()
+                state: tp.Dict[str, tp.Any] = {"fitted_checkpoint": checkpoint}
```

Comment:
Please do it outside of file context manager

Reply:
fixed